Settings provider

Brunettow opened the pull request:
Provider for Settings Preferences Implemented
Provider for Change Password Implemented

- Dummy token is used, will be changed after token part is implemented
- Actions due to response messages not handled

omerfaunal commented on project/FrontEnd/collaborative_science_platform/lib/providers/settings_provider.dart:
```diff
@@ -0,0 +1,56 @@
+
+class SettingsProvider with ChangeNotifier {
+  String token = "cd018223f0564aa0f66af496d754c819494de87f"; //TEST
+  Future<void> changePassword(String oldPass, String newPass) async {
```
Here changePassword should take User user as argument.
changePassword(User user, String oldPass, String newPass)

omerfaunal commented on project/FrontEnd/collaborative_science_platform/lib/providers/settings_provider.dart:
```diff
@@ -0,0 +1,56 @@
+    final Map<String, String> header = {
+      "Accept": "application/json",
+      "content-type": "application/json",
+      'Authorization': 'Token $token',
```
And here we should use user.token 

Before merging let's fix this. 

I know instead you used dummy token but still this is not a good usage. Instead setting user.token to something dummy is a better approach. 

For now don't add anything dummy, just add user.

omerfaunal commented on project/FrontEnd/collaborative_science_platform/lib/providers/settings_provider.dart:
```diff
@@ -0,0 +1,56 @@
+    }
+  }
+
+  Future<void> changePreferences(String bio, bool sendNotification, bool showActivity) async {
```
Same here. User user

omerfaunal commented on project/FrontEnd/collaborative_science_platform/lib/providers/settings_provider.dart:
```diff
@@ -0,0 +1,56 @@
+    final Map<String, String> header = {
+      "Accept": "application/json",
+      "content-type": "application/json",
+      'Authorization': 'Token $token',
```
user.token

omerfaunal commented on project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/widgets/account_settings_form.dart:
```diff
@@ -77,6 +78,13 @@ class _AccountSettingsFormState extends State<AccountSettingsForm> {
+  void changePreff() async {
+    
+      final settingsProvider = Provider.of<SettingsProvider>(context,listen: false);
+      await settingsProvider.changePreferences(aboutMeController.text, isSwitched, isSwitched2); 
```
Provider.of<Auth>(context, listen: false).user;

omerfaunal commented on project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/widgets/change_password_form.dart:
```diff
@@ -15,23 +17,37 @@ class ChangePasswordForm extends StatefulWidget {
+  void changePass() async {
+    
+      if (currentPass != "" && oldPassController.text != "") {
+        final settingsProvider = Provider.of<SettingsProvider>(context,listen: false);
```
Provider.of(context, listen: false).user;

Brunettow commented:
Thank you for all the comments. I changed the code accordingly. I also added error and response messages for creating more a responsive page. There are also minor design changes.
